tcms/utils/permissions.py

```diff
@@ -3,30 +3,46 @@
 from django.contrib.auth.models import Group, Permission
 
 
-def assign_default_group_permissions():
+def assign_default_group_permissions(output=None, refresh_all=False):
     """
         Adds the default permissions for Administrator and Tester
         groups!
     """
     admin = Group.objects.get(name='Administrator')
-    if admin.permissions.count() == 0:
+    if output:
+        output.write('\nAdding permissions to Administrator group')
+    if admin.permissions.count() == 0 or refresh_all:
         all_perms = Permission.objects.all()
-        admin.permissions.add(*all_perms)
+        admin_perms = admin.permissions.all()
+        perms_to_add = all_perms.exclude(pk__in=admin_perms)
+        admin.permissions.add(*perms_to_add)
+        admin.save()
+        if output:
+            for perm in perms_to_add:
+                output.write('%s.%s added' % (perm.content_type.app_label,
+                                              perm.codename))
+            output.write('Done.')
 
     tester = Group.objects.get(name='Tester')
-    if tester.permissions.count() == 0:
+    tester_perms = tester.permissions.all()
+    if output:
+        output.write('\nAdding permissions to Tester group')
+    if tester.permissions.count() == 0 or refresh_all:
         # apply all permissions for test case & product management
         for app_name in ['bugs', 'django_comments', 'linkreference', 'management',
-                         'testcases', 'testplans', 'testruns']:
-            app_perms = Permission.objects.filter(content_type__app_label__contains=app_name)
-            tester.permissions.add(*app_perms)
-
-    # this app was introduced later and we don't want all of its permissions
-    if tester.permissions.filter(content_type__app_label='attachments').count() == 0:
-        attachment_perms = Permission.objects.filter(
-            content_type__app_label='attachments'
-        ).exclude(codename='delete_foreign_attachments')
-        tester.permissions.add(*attachment_perms)
+                         'testcases', 'testplans', 'testruns', 'attachments']:
+            app_perms = Permission.objects.filter(
+                content_type__app_label__contains=app_name)
+            perms_to_add = app_perms.exclude(pk__in=tester_perms)
+            perms_to_add = perms_to_add.exclude(content_type__app_label='attachments',
+                                                codename='delete_foreign_attachments')
+            tester.permissions.add(*perms_to_add)
+            if output:
+                for perm in perms_to_add:
+                    output.write('%s.%s added' % (perm.content_type.app_label,
```

Comment:
same comment for the information message. Needs to show which group was modified.